heltour/tournament/views.py

```diff
@@ -1839,14 +1840,16 @@ def view(self, post=False):
                     for p in player_list:
                         field_name = 'av_r%d_%s' % (r.number, p.lichess_username)
                         is_available = self.request.POST.get(field_name) != 'on'
-                        if is_available != is_available_dict.get((r.id, p.id), True):
+                        
+                        if (is_available != is_available_dict.get((r.id, p.id), True) and 
+                        SeasonPlayer.objects.filter(player=p, season=self.season).first().card_color != "red"):
```

Comment:
`first()`  will sometimes return `None`. Are we sure that this will never error?

Reply:
It _should_ not error from the context when the view is called. But I will submit a commit that handles it.